force-app/main/default/classes/JobApplicationTaskCreation.cls

```
@@ -0,0 +1,164 @@
+
+        // Decide which applications should get tasks
+        List<Job_Application__c> toProcess = new List<Job_Application__c>();
+        // Loop through all the job applications in the new list of job applications that came from the trigger.new
+        for (Job_Application__c ja : newList) {
+            // Check if status is null then statusNow is null, otherwise trim status of whitespace to be safe
+            String statusNow = (ja.Status__c == null) ? null : ja.Status__c.trim();
+            // If statusNow is blank or taskByStatus map doesn't contain the same status found in statusNow jump to next loop iteration
+            if (String.isBlank(statusNow) || !tasksByStatus.containsKey(statusNow)) continue;
+
+            // On insert, always proces; on update, only if status actually changed
+            if (oldMap == null) {
+                toProcess.add(ja);
+            } else {
+                Job_Application__c oldJa = oldMap.get(ja.Id);
+                String statusOld = (oldJa == null || oldJa.Status__c == null) ? null : oldJa.Status__c.trim();
+                if (statusOld != statusNow) {
+                    toProcess.add(ja);
+                } 
+            }
+        }
+        // Empty list, end
+        if(toProcess.isEmpty()) return;
+
+        // WhatIds and subjects to prevent duplicates
+        Set<Id> whatIds = new Set<Id>();
+        Set<String> subjectsNeeded = new Set<String>();
+        // Map from each job app Id to it's planned list of TaskContainer items
+        Map<Id, List<TaskContainer>> planPerApp = new Map<Id, List<TaskContainer>>();
+
+        // Loop through each of the job applications in the toProcess list
+        for (Job_Application__c ja : toProcess) {
+            // Put the job application Id in the whatIds set
+            whatIds.add(ja.Id);
+            // Get task plan for the job apps current status
+            List<TaskContainer> plan = tasksByStatus.get(ja.Status__c.trim());
```

I would check explicitly to make sure plan isn't empty and confirm a task plan is configured for the given status. If not, you can surface a message explaining. Otherwise, the code will just silently continue